pkg/storage/stores/shipper/bloomshipper/store.go

```diff
@@ -265,6 +267,13 @@ func (b *bloomStoreEntry) Client(_ model.Time) (Client, error) {
 	return b.bloomClient, nil
 }
 
+// Allocator implements Store.
+// While bloomStoreEntry implements the Store interface, this method must never
+// be used directly and therefore can safely return nil.
+func (b *bloomStoreEntry) Allocator() mempool.Allocator {
+	return nil
```

Comment:
nit: I think would be better to no having to satisfy the interface in this way. Maybe we can have 
```go
type StoreWithAllocator interface {
    Store
    Allocator() mempool.Allocator
}
```

Ans only `BloomStore` implements that.

Having said that, if we don't want to do this, I think we should rather panic here.

Reply:
Good idea. I will add the interface.